### PR TITLE
Implement ForceFlush spec adding a timeout parameter.

### LINKED ISSUE #186: Implement ForceFlush spec in built-in Span processors
As specified here:
[https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#forceflush-1](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#forceflush-1)

Missing implementing a timeout in the protocol, and return if success. Do not break existing API, and add deprecation notice.

Also Exporters `forceFlush` are missing timeout parameter

### PR DESCRIPTION
**Pull request by nachoBonafonte**
This parameter takes effect only in BatchSpanProcessor built-in processors

Fixes #186

### CODE REVIEW

File: Sources/OpenTelemetrySdk/Trace/SpanProcessor.swift [hunk 1]
```diff
@@ -33,5 +33,12 @@ public protocol SpanProcessor {
 
     /// Processes all span events that have not yet been processed.
     /// This method is executed synchronously on the calling thread
-    func forceFlush()
+    func forceFlush(timeout: TimeInterval?)
```
Comment (SergeyKanzhelev) [inline Sources/OpenTelemetrySdk/Trace/SpanProcessor.swift:36]: should the new argument be documented? Specifically, end users will appreciate some information on what does zero timeout means and how to make it infinite

File: Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift [hunk 2]
```diff
@@ -109,33 +109,33 @@ private class BatchWorker: Thread {
...
-    private func exportBatch(spanList: [ReadableSpan]) {
+    private func exportBatch(spanList: [ReadableSpan], explicitTimeout: TimeInterval?) {
         let exportOperation = BlockOperation { [weak self] in
             self?.exportAction(spanList: spanList)
         }
         let timeoutTimer = DispatchSource.makeTimerSource(queue: DispatchQueue.global())
-        timeoutTimer.setEventHandler{ exportOperation.cancel() }
-        timeoutTimer.schedule(deadline: .now() + .milliseconds(Int(exportTimeout.toMilliseconds)), leeway: .milliseconds(1))
+        timeoutTimer.setEventHandler { exportOperation.cancel() }
+        timeoutTimer.schedule(deadline: .now() + .milliseconds(Int(explicitTimeout?.toMilliseconds ?? exportTimeout.toMilliseconds)), leeway: .milliseconds(1))
```
Comment (SergeyKanzhelev) [inline Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift:138]: is timeout that is higher than exportTimoue make sense?

File: Sources/OpenTelemetrySdk/Trace/SpanProcessor.swift (hunk 1 above)
Comment (nachoBonafonte) [reply Sources/OpenTelemetrySdk/Trace/SpanProcessor.swift:38]: Of course it should, added

File: Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift (hunk 2 above)
Comment (nachoBonafonte) [reply Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift]: Yes, it really doesn't make much sense, it will use the lower of them

**Pull request merged**